src/material.cpp

```diff
@@ -2013,6 +2013,20 @@ std::vector<std::pair<double, double> > pyne::Material::normalize_radioactivity(
   return normed;
 }
 
+void pyne::Material::from_activity(std::map<int, double> activities) {
+  // activities must be of the form {nuc: act}, eg, tritium
+  //  10030: 1.0
+
+  // clear existing components
+  comp.clear();
+
+  for (std::map<int, double>::iterator acti = activities.begin(); acti != activities.end(); acti++) {
+    comp[acti->first] = (acti->second) * pyne::atomic_mass(acti->first) / \
```

Comment:
One of our strategies is for the C++ code to behave identically to the python code.  It is therefore preferred to move the test for non-zero decay constant to here rather than in the python.